jina/drivers/convert.py

```diff
@@ -134,8 +132,23 @@ def png_pack(png_tag, data):
             png_pack(b'IHDR', struct.pack('!2I5B', self.width, self.height, 8, 6, 0, 0, 0)),
             png_pack(b'IDAT', zlib.compress(raw_data, 9)),
             png_pack(b'IEND', b'')])
-        d.uri = 'data:image/png;base64,' + base64.b64encode(png_bytes).decode()
+    
+        return 'data:image/png;base64,' + base64.b64encode(png_bytes).decode()
 
+    def convert(self, arr):
+        arr.uri = self.png_convertor(arr)
+
+
+class Blob2PngURI(NdArray2PngURI):
+    """Simple DocCrafter used in :command:`jina hello-world`,
+        it reads ``buffer`` into base64 png and stored in ``uri``"""
+
+    def __init__(self, target='uri', width: int = 28, height: int = 28, *args, **kwargs):
+        super().__init__(target, width, height, *args, **kwargs)
+
+    def convert(self, d):
+        arr = pb2array(d.blob)
+        d.uri = super().png_convertor(arr)
```

Comment:
This can now be `self.png_convertor(arr)`. No need for the `super()` magic anymore.

Reply:
done